sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs

```diff
@@ -2679,6 +2679,54 @@ async Task ProcessMessage(ProcessSessionMessageEventArgs args)
             }
         }
 
+        [Test]
+        public async Task SessionOrderingIsGuaranteedProcessor()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: true))
+            {
+                await using var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+                long lastSequenceNumber = 0;
+
+                await using var processor = client.CreateSessionProcessor(scope.QueueName,
+                    new ServiceBusSessionProcessorOptions
+                    {
+                        MaxConcurrentCallsPerSession = 1, MaxConcurrentSessions = 1
+                    });
+                processor.ProcessMessageAsync += ProcessMessage;
+                processor.ProcessErrorAsync += SessionErrorHandler;
+
+                var sender = client.CreateSender(scope.QueueName);
+
+                CancellationTokenSource cts = new CancellationTokenSource();
+                cts.CancelAfter(TimeSpan.FromSeconds(60));
+                await processor.StartProcessingAsync();
+                await SendMessagesAsync();
+
+                await processor.StopProcessingAsync();
+
+                async Task SendMessagesAsync()
+                {
+                    while (!cts.IsCancellationRequested)
+                    {
+                        await sender.SendMessageAsync(ServiceBusTestUtilities.GetMessage("session"));
+                        await Task.Delay(TimeSpan.FromMilliseconds(100));
+                    }
+                }
+
+                Task ProcessMessage(ProcessSessionMessageEventArgs args)
+                {
+                    if (args.Message.SequenceNumber != lastSequenceNumber + 1)
```

Comment:
nit:  Couldn't we do the entire if as an Assert clause?  Something like:

```csharp
Assert.That(args.Message.SequenceNumber, Is.EqualTo(lastSequenceNumber + 1), $"Last sequence number: {lastSequenceNumber}, current sequence number: {args.Message.SequenceNumber}");
```

Reply:
Hah, I don't really use that assertion style anywhere in this file, but I suppose you are right.

Reply:
Not meant to suggest that exact style, just what I'm used to writing.  `Assert.Equals` would do the same thing, I think.

Reply:
Ah, right, yeah I picked up on that after my comment, but I still copied your code verbatim.